bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/model/Eclipse.java

```diff
@@ -12,52 +12,73 @@
  */
 package org.openhab.binding.astro.internal.model;
 
+import java.util.AbstractMap.SimpleEntry;
 import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
 
-import org.apache.commons.lang.builder.ToStringBuilder;
-import org.apache.commons.lang.builder.ToStringStyle;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 
 /**
  * Holds eclipse informations.
  *
  * @author Gerhard Riegler - Initial contribution
  */
+@NonNullByDefault
 public class Eclipse {
-    private Calendar total;
-    private Calendar partial;
+    private final Map<EclipseKind, Entry<Calendar, @Nullable Double>> entries = new HashMap<>();
 
-    /**
-     * Returns the date of the next total eclipse.
-     */
-    public Calendar getTotal() {
-        return total;
+    public Eclipse(List<EclipseKind> eclipses) {
+        eclipses.forEach(eclipseKind -> {
+            entries.put(eclipseKind, new SimpleEntry<>(Calendar.getInstance(), null));
+        });
+    }
+
+    public Set<EclipseKind> getKinds() {
+        return entries.keySet();
     }
 
     /**
-     * Sets the date of the next total eclipse.
+     * Returns the date of the next total eclipse.
      */
-    public void setTotal(Calendar total) {
-        this.total = total;
+    public @Nullable Calendar getTotal() {
+        return entries.get(EclipseKind.TOTAL).getKey();
```

Comment:
You need to make sure that the map doesn't hold a null value at that key, otherwise you will get a NPE when calling `getKey()`. The same null check should be done for the other getter methods as well.